### PR TITLE
Add settings, onlyTestnet, routeConfig, filter props

### PR DESCRIPTION
**Pull request by toddkao**
Add settings prop
- slippage
- customGasAmount

Add custom slippage input
Screenshot 2024-10-28 at 4 38 41 PM

Add onlyTestnet
Add routeConfig
Add filter

### CODE REVIEW

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGetFilteredChains.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        return chain ? { ...chain, asset } : null;
+      })
+      .filter((chain) => {
+        if (!chainFilter || !chainFilter[context]) return chain !== undefined;
+        const chainIdsAllowed = Object.keys(chainFilter[context]);
+        if (chainIdsAllowed && chain?.chainID) {
+          return chainIdsAllowed.includes(chain.chainID);
+        }
+        return chain !== undefined;
+      }) as ChainWithAsset[];
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGetFilteredChains.ts:39]: filtering by filter prop here for chains

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGetGroupedAssetsByRecommendedSymbol.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        const [chainId, denoms] = entries;
+        if (denoms) {
+          return denoms.includes(asset.denom) && chainId === asset.chainID;
+        } else if (chainId) {
+          return chainId === asset.chainID;
+        }
+      }));
+    }
+    return _assets;
+  }, [_assets, chainFilter, context]);
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGetGroupedAssetsByRecommendedSymbol.ts:33]: filtering by filter prop here for assets + chains

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx [hunk 3]
```diff
@@ -254,28 +124,19 @@ export const TokenAndChainSelectorModal = createModal(
...
-      }
-      return filteredChains;
-    }, [filteredAssets, filteredChains, groupedAssetSelected, modalProps.context, networkSelection]);
 
+    const listOfAssetsOrChains = useMemo(() => {
+      if (selectChain || groupedAssetSelected) {
+        return filteredChains;
+      }
+      return filteredAssets;
+    }, [filteredAssets, filteredChains, groupedAssetSelected, selectChain]);
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx:139]: simplify logic

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGroupedAssetsByRecommendedSymbol.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        const [chainId, denoms] = entries;
+        if (denoms) {
+          return denoms.includes(asset.denom) && chainId === asset.chainID;
+        } else if (chainId) {
+          return chainId === asset.chainID;
+        }
+      }));
+    }
+    return _assets;
+  }, [_assets, chainFilter, context]);
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/useGroupedAssetsByRecommendedSymbol.ts:33]: filtering by `filter` prop implemented here for assets + chains

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/useFilteredChains.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        // Check if chain is allowed based on chainFilter
+        const isAllowedByFilter = !chainFilter?.[context] ||
+          Object.keys(chainFilter[context]).includes(chain.chainID);
+
+        // For source context, exclude Penumbra chains
+        const isPenumbraAllowed = context !== "source" ||
+          !chain.chainID.includes("penumbra");
+
+        return isAllowedByFilter && isPenumbraAllowed;
+      }) as ChainWithAsset[];
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/useFilteredChains.ts:43]: filtering by `filter` prop implemented here for chains, also moved filtering out penumbra from source chains here

File: packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx [hunk 6]
```diff
@@ -41,20 +41,18 @@ export type SelectorContext = "source" | "destination";
 export type TokenAndChainSelectorModalProps = ModalProps & {
   onSelect: (token: ClientAsset | null) => void;
   selectedAsset?: ClientAsset;
-  networkSelection?: boolean;
-  context: SelectorContext
+  selectChain?: boolean;
```
Comment (toddkao) [inline packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx:44]: rename prop to `selectChain`, I think it's best to be consistent and always use the same word when referring to `chains`

Comment (toddkao) [reply packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx:44]: I'm going to refactor/change the name of this Modal to `AssetsAndChainSelectorModal` for the same reason, but I will do it in a separate PR to not mess with the diff

Comment (toddkao) [reply packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx:44]: https://github.com/skip-mev/skip-go/pull/388

File: packages/widget-v2/src/pages/SwapPage/SwapDetailModal.tsx [hunk 7]
```diff
@@ -31,7 +31,7 @@ export const SwapDetailModal = createModal((modalProps: ModalProps) => {
   const clientOperations = route && getClientOperations(route.operations);
 
   const usesEvmInOperations = useMemo(() => {
-    return clientOperations?.find(operation => operation.toChainID === "1");
+    return clientOperations?.find((operation) => operation.toChainID === "1");
```
Comment (codingki) [inline packages/widget-v2/src/pages/SwapPage/SwapDetailModal.tsx:34]: this logic is wrong, we need to check `toChainID`'s `chainType`

Comment (toddkao) [reply packages/widget-v2/src/pages/SwapPage/SwapDetailModal.tsx]: @codingki we were missing this disclaimer in SwapExecutionPageRouteDetailed so I made a component with the new logic inside

Screenshot 2024-10-29 at 12 13 39 PM

File: packages/widget-v2/src/components/EvmDisclaimer.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+  const usesEvmInOperations = useMemo(() => {
+    return operations?.find(
+      (operation) => {
+        const chainTypeIn = chains?.find(chain => chain.chainID === operation.fromChainID)?.chainType;
+        const chainTypeOut = chains?.find(chain => chain.chainID === operation.toChainID)?.chainType;
+        return chainTypeIn === "evm" || chainTypeOut === "evm";
+      }
+    );
+  }, [chains, operations]);
```
Comment (toddkao) [inline packages/widget-v2/src/components/EvmDisclaimer.tsx:20]: updated `usesEvmInOperations` logic and made a component so we can reuse

**Review by codingki — APPROVED** [review]

**Pull request merged**